test/unit/iamService.ts

```diff
@@ -0,0 +1,146 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ** This file is automatically generated by gapic-generator-typescript. **
+// ** https://github.com/googleapis/gapic-generator-typescript **
+// ** All changes to this file may be overwritten. **
+import * as assert from 'assert';
+import {describe, it} from 'mocha';
+import {GrpcClient as FallbackGrpcClient} from '../../src/fallback';
+import {GrpcClient} from '../../src/grpc';
+import {IamClient, IamProtos} from '../../src';
+
+class FakeError {
+  name: string;
+  message: string;
+  code: number;
+  constructor(n: number) {
+    this.name = 'fakeName';
+    this.message = 'fake message';
+    this.code = n;
+  }
+}
+export interface Callback {
+  (err: FakeError | null, response?: {} | null): void;
+}
+function mockSimpleGrpcMethod(
```

Comment:
Can we use `sinon`-style mocks here as in the generated unit tests?

Reply:
done.